### PR TITLE
Add purchase page logic

### PR DESCRIPTION
**Pull request by gopipatell**

### CODE REVIEW

File: src/components/PurchasePage/Order/index.tsx [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+  const dispatch = useDispatch();
+  const clearAllProductsAction = () => {
+    dispatch(clearAllProducts());
+  };
+  const products: Array<IProductSelector> = useSelector(productsSelector);
+  const subtotalAmount: number = useSelector(productSubTotalSelector);
+
+  const [surchargeAmount, setSurchargeAmount] = useState(0);
+  const [tipAmount, setTipAmount] = useState(0);
+  const [cashoutAmount, setCashoutAmount] = useState(0);
```
Comment (mattgresham) [inline src/components/PurchasePage/Order/index.tsx:24]: I'm assuming the `useState`'s are temporary until you get to the keypad?

I like the selectors!

File: src/components/PurchasePage/Order/index.tsx [hunk 2]
```diff
@@ -14,23 +39,37 @@ function Order(): React.ReactElement {
           Order
         </Typography>
         <Box>
-          <Button className={classes.clear}>Clear all</Button>
+          <Button className={classes.clear} onClick={() => clearAllProductsAction()}>
```
Comment (mattgresham) [inline src/components/PurchasePage/Order/index.tsx:42]: More an FYI, than a request, but if you wanted to with handlers like this, you can pass just the reference to the function without the invocation part:

```suggestion
          <Button className={classes.clear} onClick={clearAllProductsAction}>
```

**Review by mattgresham — COMMENTED** [review]
Nice work! Great to see it come alive!

File: src/components/PurchasePage/Order/index.tsx (hunk 1 above)
Comment (gopipatell) [reply src/components/PurchasePage/Order/index.tsx:24]: yes I am working on the slice will do it in next PR

**Review by mattgresham — APPROVED** [review]
Nice work! Particularly on the formatting helper and the selectors!

**Pull request merged**